helm-chart/renku/values.yaml

```diff
@@ -126,6 +136,23 @@ global:
       tag: "0.0.2"
     customCAs: []
       # - secret:
+  
+  ## Values for the RabbitMQ setup.
+  rabbitmq:
+    ## The external section is relevant only if an external RabbitMQ will be used
+    ## with Renku. The RabbitMQ management plugin needs to be enabled so that the
+    ## proper vhost, users and permissions are setup.
+    external:
+      disabled: true
```

Comment:
@rokroskar I wanted to not pollute the `rabbitmq` "namespace" in the values file. I know we do this for other charts but I think it is something we should move away from. By adding `rabbitmq.enabled` in the values file we make this available to the internal/child chart and if the child chart ends up validating this in a specific way or using it in some way then things will break. Can I instead do `global.rabbitmq.internal.enabled`? This way the option to enable/disable the internal chart is not part of the internal chart itself.

That is the main reason I wanted to deviate from our current practices. Ideally we should avoid adding any values in child helm charts that are actually not part of that chart originally.

Reply:
that makes sense. Maybe not for this PR, but maybe having something like `global.thirdParty.rabbitmq.enabled`? We can follow that for all the third-party charts we use. Maybe too verbose. But aren't you polluting the namespace with `external` or is that something that rabbitmq is already using?

Reply:
@rokroskar sure I can use `global.thirdParty.rabbitmq.enabled`. But adding `global.rabbitmq` does not pollute the rabbitmq helm chart. Looking at the [values file](https://github.com/bitnami/charts/blob/main/bitnami/rabbitmq/values.yaml) for the rabbitmq bitnami helm chart it only uses `global.imageRegsitry`, `global.imagePullSecret` and `global.storageClass`. So we are not polluting anything by using `global.rabbitmq`. I think this is the case with all other third party helm charts we use too. They do not really use `global` at all. So that is a good stop to put all the Renku configuration for such 3rd party components.

Reply:
oh ok, so `global.<service>.enabled` seems like a reasonable pattern?

Reply:
Whatever we decide to do here, I will follow suit with Postgres in #2122

Reply:
Do we need `global.<service>.enabled`? Can't we just derive that from `rabbitmq.enabled`? If one is true, the other must be false?

I have been at odds with this exact topic in #2122 in regards to Postgres. However we decide to deal with externally provided services, I want to use that same approach for both rabbitmq and postgres

Reply:
Sorry for unresolving this but the structure for defining external third-party service isn't quite clear to me yet.

Reply:
@wesjdj sorry I missed your comment before and saw it just now when looking at the whole PR again. See the [comment](https://github.com/SwissDataScienceCenter/renku/pull/2872#discussion_r1054504991) from @olevski above. The main point of `global.<service>.enabled` is to avoid putting _anything at all_ in the namespace of the external service to avoid conflicts. But yes, as you suggest, lets write down the guidelines and make sure this PR can conform.